packages/gcp/src/services/storage.ts

```diff
@@ -0,0 +1,661 @@
+"use strict";
+import { Storage as GCS, GetSignedUrlConfig } from "@google-cloud/storage";
+import {
+  Binary,
+  BinaryMap,
+  BinaryParameters,
+  Context,
+  CoreModel,
+  ModdaDefinition,
+  EventBinaryGet,
+  WebdaError,
+  BinaryFile,
+} from "@webda/core";
+import { Readable, Stream } from "stream";
+import { join } from "path";
+
+export type StorageObject = {
+  key: string;
+
+  /**
+   * bucket is an option,
+   * if not provided: it will get the DEFAULT_BUCKET set from environment variable
+   */
+  bucket?: string;
+};
+
+export type DownloadParams = StorageObject & {
+  localPath: string;
+};
+
+export type PutObjectParams = StorageObject & {
+  localPath?: string;
+  content?: string | Buffer;
+};
+
+export type GetSignedUrlParams = StorageObject & {
+  expires?: number;
+  action: "read" | "write" | "delete";
+};
+
+export type StorageObjectMeta = {
+  size: number;
+  contentType: string;
+};
+
+export class StorageParameters extends BinaryParameters {
+  endpoint?: string;
+  prefix?: string;
+  CloudFormation: any;
+  CloudFormationSkip: boolean = false;
+
+  /**
+   * Bucket
+   */
+  private _bucket?: string;
+  get bucket(): string {
```

Comment:
the loader of parameters does `Object.assign(this, params)`, I do not think we want to have the webda.config.json to have a parameter `_bucket`

Reply:
removed